moviebot/controller/controller_flask.py

```diff
@@ -0,0 +1,398 @@
+"""This file contains the Controller class which controls the flow of the
+conversation while the user interacts with the agent using Messenger."""
+
+import json
+import os
+import sqlite3
+from datetime import datetime
+from typing import Any, Callable, Dict, Union
+
+from moviebot.agent.agent import Agent
+from moviebot.controller.controller import Controller
+from moviebot.controller.http_data_formatter import (
+    HTTP_OBJECT_MESSAGE,
+    HTTPDataFormatter,
+)
+from moviebot.core.utterance.utterance import UserUtterance
+
+
+class ControllerFlask(Controller):
+    def __init__(self) -> None:
+        """Initializes structs for Controller and sends the get started button
+        to the client."""
+        self.token = ""
+        self.agent = {}
+        self.record_data = {}
+        self.user_messages = {}
+        self.record_data_agent = {}
+        self.user_options = {}
+        self.agent_response = {}
+        self.configuration = {}
+        self.info = {}
+        self.users = {}
+        self.load_data = {}
+        self.path = ""
+        self.methods = [
+            {"payload": "/start", "action": self.first_time_message},
+            {"payload": "/help", "action": self.instructions},
+            {"payload": "/accept", "action": self.store_user},
+            {"payload": "/reject", "action": self.start_conversation},
+            {"payload": "/restart", "action": self.restart},
+            {"payload": "/exit", "action": self.exit},
+            {"payload": "/delete", "action": self.delete_data},
+            {"payload": "/test", "action": self.privacy_policy},
+            {"payload": "/store", "action": self.store_user},
+            {"payload": "/continue", "action": self.start_conversation},
+        ]
+        self.agent_intent = ""
+
+    def lookup(self) -> sqlite3.Cursor:
+        """Returns SQL cursor."""
+        conn = sqlite3.connect(self.configuration["DATA"]["db_path"])
+        c = conn.cursor()
+        return c
+
+    def initialize(self, user_id: str) -> None:
+        """Initializes structs for a new user.
+
+        Args:
+            user_id: User id.
+        """
+        if user_id not in self.agent:
+            self.user_options[user_id] = {}
+            self.users[user_id] = {}
+            self.user_messages[user_id] = HTTPDataFormatter(user_id)
```

Comment:
Do we really need to store a separate instance of HTTPDataFormatter for each user? Consider making it user agnostic.

Reply:
I removed the class `HTTPDataFormatter` (PR #125) and the property `user_messages` as it is not needed anymore.